src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -128,7 +128,19 @@ type ClusterSpec struct {
 	// List of superusers
 	Superusers []Superuser `json:"superUsers,omitempty"`
 	// SASL enablement flag
+	// +optional
+	// Deprecated: replaced by "kafkaEnableAuthorization"
 	EnableSASL bool `json:"enableSasl,omitempty"`
+	// Enable authorization for Kafka connections. Values are:
+	//
+	// - `nil`: Ignored. Authorization is enabled with `enable_sasl: true`
+	//
+	// - `true`: authorization is required
+	//
+	// - `false`: authorization is disabled;
+	//
+	// See also `enableSasl` and `configuration.kafkaApi[].authenticationMethod`
+	KafkaEnableAuthorization bool `json:"kafkaEnableAuthorization,omitempty"`
```

Comment:
why do we need to have this global flag? Why `configuration.kafkaApi[].authenticationMethod` is not enough? 🤔